src/platform/plugins/shared/discover/public/application/main/state_management/redux/internal_state.ts

```diff
@@ -130,15 +111,6 @@ export const internalStateSlice = createSlice({
       state.tabs.recentlyClosedTabIds = action.payload.recentlyClosedTabs.map((tab) => tab.id);
     },
 
-    setDataViewId: (state, action: TabAction<{ dataViewId: string | undefined }>) =>
```

Comment:
The `tab.dataViewId` value was never actually used so I removed it to avoid confusion and duplication.